### PR TITLE
CRUD for collection item template

### PR DESCRIPTION
**Pull request by jonas-atmire**
REST API implementation for the rest contract changes mentioned in:

* DSpace/Rest7Contract#80
This has recently been slightly altered by the following as well
* DSpace/Rest7Contract#85

#### Quick overview:
This extends the codebase with the CRUD functionality for the collection item template

### CODE REVIEW

**Review by KevinVdV — CHANGES_REQUESTED** [review]
Made an initial review & found a few small things that require changing.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/CollectionItemtemplateController.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     */
+    @PreAuthorize("hasPermission(#uuid, 'COLLECTION', 'WRITE')")
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<ResourceSupport> createTemplateItem(HttpServletRequest request,
+                                                              @PathVariable UUID uuid,
+                                                              @RequestBody(required = false) JsonNode itemBody)
+            throws SQLException, AuthorizeException {
+
+        if (itemBody == null) {
+            throw new BadRequestException("The new item should be included as json in te body of this request");
```
Comment (mspalti) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/CollectionItemtemplateController.java]: Minor typo: 'the body'

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemTemplateRestControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+
+    private ObjectMapper mapper;
+    private String adminAuthToken;
+    private Collection childCollection;
+    private ItemRest testTemplateItem;
+    private String patchBody;
+
+    @Before
+    public void createStructure() throws Exception {
+        context.turnOffAuthorisationSystem();
```
Comment (mspalti) [inline dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemTemplateRestControllerIT.java]: Would it be a good idea to also restoreAuthSystemState? The tests below pass so I guess it's not necessary!

**Review by mspalti — COMMENTED** [review]
@jonas-atmire , the code looks good to me.  I added two minor comments.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemTemplateRestControllerIT.java (hunk 2 above)
Comment (jonas-atmire) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/ItemTemplateRestControllerIT.java]: @mspalti As the tests are 'reset' to a general predefined stats, as per this setup method: https://github.com/DSpace/DSpace/blob/6e56c8bad15274da523b5f6a819b8f67030ee541/dspace-server-webapp/src/test/java/org/dspace/app/rest/test/AbstractIntegrationTestWithDatabase.java#L98-L106, there is no actual need to do this.

**Review by mspalti — APPROVED** [review]

**Review by KevinVdV — APPROVED** [review]
I would just like to request 2 small tweaks to variables that are duplicated, once these are removed I approve.

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java [hunk 3]
```diff
@@ -70,9 +70,14 @@
     @Autowired
     private BitstreamService bitstreamService;
 
+    @Autowired
+    private ItemConverter itemConverter;
```
Comment (KevinVdV) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java]: This variable is now unused & can be removed.

**Review by KevinVdV — APPROVED** [review]
There is just one small thing left before I can merge this PR, with the latest merge in master a single variable has become unused. Remove it & I can merge it.

**Pull request merged**